packages/ra-core/src/dataProvider/useDelete.ts

```diff
@@ -162,8 +162,8 @@ export const useDelete = <
             meta: callTimeMeta = paramsRef.current.meta,
         } = {}) =>
             dataProvider
-                .delete<RecordType>(callTimeResource, {
-                    id: callTimeId,
+                .delete<RecordType>(callTimeResource as string, {
+                    id: callTimeId as Identifier,
```

Comment:
Similar to the hack above, and we have similar options: 1) `callTimeId` should only be optional if `params.id` is not undefined, and required otherwise 2) just do a runtime check